src/integration_fleets_tutorial_cpp.md

```diff
@@ -0,0 +1,347 @@
+    *battery_system, mechanical_system);
+
+  // Ambient power system
+  const YAML::Node ambient_system = rmf_fleet["ambient_system"];
+  const double ambient_power_drain = ambient_system["power"].as<double>();
+  auto ambient_power_system = rmf_battery::agv::PowerSystem::make(
+    ambient_power_drain);
+  if (!ambient_power_system)
+  {
+    RCLCPP_ERROR(
+      node->get_logger(),
+      "Invalid values supplied for ambient power system");
+
+    return false;
+  }
+  std::shared_ptr<rmf_battery::agv::SimpleDevicePowerSink> ambient_sink =
+    std::make_shared<rmf_battery::agv::SimpleDevicePowerSink>(
+    *battery_system, *ambient_power_system);
+
+  // Tool power system
+  const YAML::Node tool_system = rmf_fleet["tool_system"];
+  const double tool_power_drain = ambient_system["power"].as<double>();
+  auto tool_power_system = rmf_battery::agv::PowerSystem::make(
+    tool_power_drain);
+  if (!tool_power_system)
+  {
+    RCLCPP_ERROR(
+      node->get_logger(),
+      "Invalid values supplied for tool power system");
+
+    return false;
+  }
+  std::shared_ptr<rmf_battery::agv::SimpleDevicePowerSink> tool_sink =
+    std::make_shared<rmf_battery::agv::SimpleDevicePowerSink>(
+    *battery_system, *tool_power_system);
+
+  // Drain battery
+  const bool drain_battery = rmf_fleet["account_for_battery_drain"].as<bool>();
+  // Recharge threshold
+  const double recharge_threshold =
+    rmf_fleet["recharge_threshold"].as<double>();
+  // Recharge state of charge
+  const double recharge_soc = rmf_fleet["recharge_soc"].as<double>();
+
+```
+
+- Setting the fleet's finishing request
+
+```cpp
+
+ // Finishing tasks
+  const YAML::Node task_capabilities = rmf_fleet["task_capabilities"];
+  const std::string finishing_request_string =
+    task_capabilities["finishing_request"].as<std::string>();
+  rmf_task::ConstRequestFactoryPtr finishing_request = nullptr;
+  if (finishing_request_string == "charge")
+  {
+    finishing_request =
+      std::make_shared<rmf_task::requests::ChargeBatteryFactory>();
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is configured to perform ChargeBattery as finishing request");
+  }
+  else if (finishing_request_string == "park")
+  {
+    finishing_request =
+      std::make_shared<rmf_task::requests::ParkRobotFactory>();
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is configured to perform ParkRobot as finishing request");
+  }
+  else if (finishing_request_string == "nothing")
+  {
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is not configured to perform any finishing request");
+  }
+  else
+  {
+    RCLCPP_WARN(
+      node->get_logger(),
+      "Provided finishing request [%s] is unsupported. The valid "
+      "finishing requests are [charge, park, nothing]. The task planner will "
+      " default to [nothing].",
+      finishing_request_string.c_str());
+  }
+
+```
+
+- Adding task capabilities and performable actions to the fleet
```

Comment:
Separate task capabilities and performable actions
@thedevmanek instead of directly applying the suggestion, could you do this from the code? I can't seem to add the closing ``` in this comment, might mess up the markdown generated
```suggestion
Users can now set the initialized task planner parameters with the fleet handle using `FleetUpdateHandle::set_task_planner_params`.

```cpp
_fleet_handle->set_task_planner_params(
    battery_system,
    motion_sink,
    ambient_sink,
    tool_sink,
    recharge_threshold,
    recharge_soc,
    account_for_battery_drain,
    finishing_request
)
```
```